code/ui/blocks/src/blocks/Canvas.stories.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import { Canvas } from './Canvas';
+import { Story as StoryComponent } from './Story';
+import * as BooleanStories from '../controls/Boolean.stories';
+
+const meta: Meta<typeof Canvas> = {
+  component: Canvas,
+  parameters: {
+    relativeCsfPaths: ['../controls/Boolean.stories'],
+  },
+  render: (args) => {
+    return (
+      <Canvas {...args}>
+        <StoryComponent of={BooleanStories.Undefined} />
+      </Canvas>
+    );
+  },
+};
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const BasicStory: Story = {};
+
+export const WithSourceOpen: Story = {
+  args: {
+    withSource: 'open',
+  },
+};
+export const WithSourceClosed: Story = {
+  args: {
+    withSource: 'closed',
+  },
+};
+
+// TODO: what is the purpose of mdxSource exactly?
+export const WithMdxSource: Story = {
```

Comment:
The `mdxSource` prop should be deprecated and renamed to something that falls in line with the naming in parameters, so `source` or `code`. I haven't quite figured out what the difference between the two yet.

any thoughts @tmeasday @shilman ?